api/src/main/java/org/openmrs/api/impl/ConceptServiceImpl.java

```diff
@@ -1982,6 +1984,72 @@ public boolean hasAnyConceptAttribute(ConceptAttributeType conceptAttributeType)
 		return dao.getConceptAttributeCount(conceptAttributeType) > 0;
 	}
 
+	/**
+	 * @see org.openmrs.api.ConceptService#getConceptByNameOrIdOrMap(String conceptRef
+	 */
+	@Override
+	@Transactional(readOnly = true)
+	public Concept getConceptByNameOrIdOrMap(String conceptRef) throws APIException {
+		if (conceptRef == null) {
+			return null;
+		}
+		Concept cpt = null;
+		//check if input is a valid Uuid
+		if (!(conceptRef.length() < 36 || conceptRef.length() > 38 || conceptRef.contains(" ") || conceptRef.contains("."))) {
+			cpt = Context.getConceptService().getConceptByUuid(conceptRef);
+			if (cpt != null)
+				return cpt;
+		}
+		//check if input is map
+		if (conceptRef.indexOf(":") != -1) {
+			int idx = conceptRef.indexOf(":");
+			String conceptSource, conceptCode;
+			conceptCode = conceptRef.substring(idx + 1);
+			conceptSource = conceptRef.substring(0, idx);
+			cpt = Context.getConceptService().getConceptByMapping(conceptCode, conceptSource);
+			if (cpt != null)
+				return cpt;
+		}
+		//check if input is a name or id
+		if (!conceptRef.isEmpty()) {
```

Comment:
This is redundant or, at the very least, should be done *before* we get here, since none of these tests are very functional if the `conceptRef` is empty.

Reply:
This must be done at the start